qlty-cli/src/commands/plugins/enable.rs

```diff
@@ -47,10 +47,24 @@ impl ConfigDocument {
             self.document["plugin"] = array();
         }
 
-        for plugin in self.document["plugin"].as_array_of_tables().unwrap() {
-            if plugin["name"].as_str() == Some(name) {
-                eprintln!("{} Plugin {} is already enabled", style("⚠").yellow(), name);
-                return Ok(());
+        if let Some(plugin_tables) = self.document["plugin"].as_array_of_tables_mut() {
+            for plugin_table in plugin_tables.iter_mut() {
+                if plugin_table["name"].as_str() == Some(name) {
+                    match plugin_table.get("mode") {
+                        Some(value) if value.as_str() == Some(IssueMode::Disabled.to_str()) => {
+                            plugin_table.remove("mode");
+                            return Ok(());
+                        }
+                        Some(_) | None => {
+                            eprintln!(
+                                "{} Plugin {} is already enabled",
```

Comment:
Is there a test for this case?

Reply:
There's a test for the `Some(_)` case but not the `None` case. Do you think it's valuable / necessary to cover both cases? I was on the fence.

Reply:
No, I think having a test for Some is sufficient, as that's the trickier case